src/store/quiz.js

```diff
@@ -0,0 +1,29 @@
+import QuizService from '@/api-services/quiz.service'
+
+const state = {
+
+}
```

Comment:
Why does your application doesn't have state? You have only one store for each application. Make use of it building quiz creator.

Imagine this situation: You have quiz creator and editor? What solution is better? To make API call every time you change your application (add quiz, x10 edit same quiz) or is it faster to store it in application state and make calls only if it is necessary.

Reply:
State of application is split between store and components. I move state to vuex only if it's reused between components.

Reply:
So tell me how you are going to do edit quiz master @Layfully

Reply:
In the future don't resolve the conversation if I didn't see your answer.

Reply:
I will fetch quiz from API, edit it and send request to patch it.

Reply:
In this case if i were to store newQuiz in store it would be of no use for editor, because it needs to work on existing quizzes. I think that storing something in vuex when it is used only by one component will lead to having big chunks of data that aren't relevant for other parts of app  which will reduce performance and I want to prevent that. In future if i will be working on component that uses same state i will move it to vuex. There is no need to future-proof that since moving state to vuex is quick and easy.

Reply:
Sorry for resolving before giving you time to reply

Reply:
Vuex uses a single state tree - that is, this single object contains all your application level state and serves as the "single source of truth". This also means usually you will have only one store for each application. A single state tree makes it straightforward to locate a specific piece of state, and allows us to easily take snapshots of the current app state for debugging purposes so on and so on. So I don't agree with:
> I think that storing something in vuex when it is used only by one component will lead to having big chunks of data that aren't relevant for other parts of app which will reduce performance and I want to prevent that. In future if i will be working on component that uses same state i will move it to vuex.

Reply:
Ok you've conviced me.